packages/flutter/test/material/dialog_test.dart

```diff
@@ -243,6 +243,36 @@ void main() {
     expect(await result, equals(42));
   });
 
+  testWidgets('Custom padding on SimpleDialogOption', (WidgetTester tester) async {
+    const EdgeInsets customPadding = EdgeInsets.all(10);
+    final SimpleDialog dialog = SimpleDialog(
+      title: const Text('Title'),
+      children: <Widget>[
+        SimpleDialogOption(
+          onPressed: () {},
+          child: const Text('First option'),
+          padding: customPadding,
+        ),
+      ],
+    );
+
+    await tester.pumpWidget(_buildAppWithDialog(dialog));
+    await tester.tap(find.text('X'));
+    await tester.pumpAndSettle();
+
+    final Size dialogSize = tester.getSize(
+      find.descendant(
+        of: find.byType(SimpleDialog),
+        matching: find.byType(Material),
+      ).first,
+    );
+    final Size optionSize = tester.getSize(
+      find.text('First option'),
+    );
+
+    expect(optionSize.width, dialogSize.width - (10.0 * 2));
```

Comment:
It might be safer to test different values for top, left, right, and bottom to ensure that each side is correctly honored. For example:
```dart
    const EdgeInsets customPadding = EdgeInsets.only(
      top: 10,
      right: 8,
      bottom: 6,
      left: 4,
    );

    // ...

    expect(optionLeftSide, dialogLeftSide + 4.0);
    expect(optionTopSide, dialogTopSide + 10.0);
    expect(optionRightSide, dialogRightSide - 6.0);
    expect(optionBottomSide, dialogBottomSide - 6.0);
```

Reply:
Good suggestion! I've changed the tests are rebased the branch with the upstream/master.